site/app/controllers/forum/ForumController.php

```diff
@@ -428,11 +428,31 @@ public function publishPost() {
                 $this->core->getNotificationFactory()->onNewPost($event);
 
                 $result['next_page'] = $this->core->buildCourseUrl(['forum', 'threads', $thread_id]) . '?' . http_build_query(['option' => $display_option]);
+                $result['post_id'] = $post_id;
+                $result['thread_id'] = $thread_id;
             }
         }
         return $this->core->getOutput()->renderJsonSuccess($result);
     }
 
+    /**
+     * @Route("/courses/{_semester}/{_course}/forum/posts/single", methods={"POST"})
+     */
+    public function getSinglePost() {
+        $post_id = $_POST['post_id'];
+        $reply_level = $_POST['reply_level'];
+        $post = $this->core->getQueries()->getPost($post_id);
+        if (($_POST['edit']) && !empty($this->core->getQueries()->getPostHistory($post_id))) {
+            $post['edit_timestamp'] = $this->core->getQueries()->getPostHistory($post_id)[0]['edit_timestamp'];
+        }
+        $thread_id = $post['thread_id'];
+        $GLOBALS['totalAttachments'] = 0;
```

Comment:
We would like to remove all usages of the $GLOBALS array.  If this isn't to much work could you include it in this PR?  If it is more trouble than its worth right now feel free to create an issue and someone can work on it later.

Reply:
in the previous version the _create_post_ function which takes the _totalAttachments_ variable passed as reference was only called in the _ForumThreadView.php_ file https://github.com/Submitty/Submitty/blob/67fedcf561cbb8bc9f9f98b30d65d41ecd1ad1ec/site/app/views/forum/ForumThreadView.php#L790
Now I'm calling this function in the _ForumController.php_ file so i couldn't pass this variable by reference from one file to another so i used global variables instead.
So we can open a new issue for this and use the global variable in the meantime.

Reply:
Let's make this a broad issue to clean up all of the usages of `$GLOBALS` in the ForumController and ForumView.